### PR TITLE
refactor(shims): port to x86_64-unknown-none

### LINKED ISSUE #1461: Port shim-sgx to x86_64-unknown-none
Currently the shim compiles for the x86_64-unknown-linux-musl target. Port it to the new x86_64-unknown-none target: https://doc.rust-lang.org/nightly/rustc/platform-support/x86_64-unknown-none.html

### LINKED ISSUE #1462: Port shim-sev to x86_64-unknown-none
Currently the shim compiles for the x86_64-unknown-linux-musl target. Port it to the new x86_64-unknown-none target: https://doc.rust-lang.org/nightly/rustc/platform-support/x86_64-unknown-none.html

### LINKED ISSUE #1574: Reduce size of shim artifacts sizes

### PR DESCRIPTION
**Pull request by bstrie**
Supersedes https://github.com/enarx/enarx/pull/1566. Thank you to @Freax13 for your assistance!

Closes #1461 
Closes #1462 
Fixes #1574

### CODE REVIEW

File: internal/shim-sgx/build.rs [hunk 1]
```diff
@@ -2,4 +2,6 @@
 
 fn main() {
     println!("cargo:rerun-if-changed=layout.ld");
+    println!("cargo:rustc-link-arg=-Tlayout.ld");
+    println!("cargo:rustc-link-arg=--sort-section=alignment");
```
Comment (haraldh) [inline internal/shim-sgx/build.rs:6]:
```suggestion
   println!("cargo:rustc-link-arg-bin=shim-sgx=-Tlayout.ld");
    println!("cargo:rustc-link-arg-bin=shim-sgx=--sort-section=alignment");
```

File: internal/shim-kvm/build.rs [hunk 2]
```diff
@@ -2,4 +2,6 @@
 
 fn main() {
     println!("cargo:rerun-if-changed=layout.ld");
+    println!("cargo:rustc-link-arg=-Tlayout.ld");
+    println!("cargo:rustc-link-arg=--sort-section=alignment");
```
Comment (haraldh) [inline internal/shim-kvm/build.rs:6]:
```suggestion
   println!("cargo:rustc-link-arg-bin=shim-kvm=-Tlayout.ld");
    println!("cargo:rustc-link-arg-bin=shim-kvm=--sort-section=alignment");
```

**Review by haraldh — CHANGES_REQUESTED** [review]

File: build.rs [hunk 3]
```diff
@@ -178,19 +183,7 @@ fn cargo_build_bin(
...
-        .arg(&out_bin)
-        .arg(&target_bin)
-        .status()?;
-
-    // Failing that, just copy it into place
-    if !status.success() {
-        println!("cargo:warning=Failed to run `strip` on {:?}", target_bin);
-        std::fs::rename(&target_bin, &out_bin)?;
-    }
+    std::fs::rename(&target_bin, &out_bin)?;
```
Comment (haraldh) [inline build.rs:186]: copy or rename? Would it mean it has to build the final binary every time again?

Comment (bstrie) [reply build.rs:186]: Good catch, although it doesn't appear to take any significant amount of time to regenerate the final binary. I'll make it copy instead.

**Review by haraldh — APPROVED** [review]

**Pull request merged**